user/TASK/chassis_task/chassis_task.c

```diff
@@ -250,6 +269,91 @@ static void increment_PID(Chassis_t *chassis_pid){
         chassis_pid->motor[BACK_LEFT].speed_read, 
         chassis_pid->motor[BACK_LEFT].current_out);
         serial_send_string(pid_out);
+        
     }
-	
+}
+
+
+
+
+/**
+* Current limiter has 4 states,
+* If over move to half
+* If over still for longer than hysteresis time, move to quarter,
+* If still over for longer than hysteresis time, move to off.
+* for all of these cases:
+* If current stays below limit for hysteresis time, reduce limitation by 1 level. 
+*/
```

Comment:
Great spec (although not what Satish would want). My question is, should this be interspersed throughout the function as comments instead? imo this reads more like all the comments stuck together and pushed above the function

Reply:
Also, this is hopefully only a temporary solution (at least the counting of executions as a timer part). Eventually I'd like to have a "judging" system like the one from Infantry_X, which would replace this.